### PR TITLE
OCPBUGS#42296: adding known issue for Azure controlplane.platform null failure

### PR DESCRIPTION
**Pull request by bscott-rh**
Version(s):
4.17

Issue:
https://issues.redhat.com/browse/OCPBUGS-42296

Link to docs preview:
https://82278--ocpdocs-pr.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-17-release-notes.html#ocp-4-17-known-issues_release-notes

QE review:
- [x] QE has approved this change.

### CODE REVIEW

File: release_notes/ocp-4-17-release-notes.adoc [hunk 1]
```diff
@@ -1883,6 +1883,15 @@ In the following tables, features are marked with the following statuses:
 
 * The DNF package manager included in {op-system-first} images cannot be used at runtime, because DNF relies on additional packages to access entitled nodes in a cluster that are under a Red Hat subscription. As a workaround, use the `rpm-ostree` command instead. (link:https://issues.redhat.com/browse/OCPBUGS-35247[*OCPBUGS-35247*])
 
+* When installing a cluster on {azure-full}, the installation will fail if the `controlPlane.platform` stanza is null. See xref:../installing/installing_azure/ipi/installing-azure-customizations.html#installation-azure-config-yaml_installing-azure-customizations[Sample customized install-config.yaml file for Azure] for a sample configuration file, or set a non-null parameter as in the following example:
```
Comment (patrickdillon) [inline release_notes/ocp-4-17-release-notes.adoc:1886]: There are two cases where it fails:
1. installing without an install config, i.e. running `openshift-install create cluster` and using the interactive commands
2. using an install config where `controlPlane` is defined but `controlPlane.Platform.Azure` is not defined.

So maybe something like 

```suggestion
* When installing a cluster on {azure-full}, the installation will fail if no install config is provided or if the `controlPlane.platform` stanza is null in the provided install-config. See xref:../installing/installing_azure/ipi/installing-azure-customizations.html#installation-azure-config-yaml_installing-azure-customizations[Sample customized install-config.yaml file for Azure] for a sample configuration file, or set a non-null parameter as in the following example:
```

[the remark this answers is not in the document]
Comment (bscott-rh) [reply release_notes/ocp-4-17-release-notes.adoc:1886]: Good catch, thx

Comment (bscott-rh) [reply release_notes/ocp-4-17-release-notes.adoc]: Updated, thx

**Comment by patrickdillon** [comment]
+1 LGTM

**Comment by sadasu** [comment]
/retitle OCPBUGS-42296: adding known issue for Azure controlplane.platform null failure

File: release_notes/ocp-4-17-release-notes.adoc [hunk 2]
```diff
@@ -1883,6 +1883,15 @@ In the following tables, features are marked with the following statuses:
 
 * The DNF package manager included in {op-system-first} images cannot be used at runtime, because DNF relies on additional packages to access entitled nodes in a cluster that are under a Red Hat subscription. As a workaround, use the `rpm-ostree` command instead. (link:https://issues.redhat.com/browse/OCPBUGS-35247[*OCPBUGS-35247*])
 
+* When installing a cluster on {azure-full}, the installation will fail if no `install-config.yaml` file is provided or if the `controlPlane.platform.azure` stanza is null in the provided `install-config.yaml` file. See xref:../installing/installing_azure/ipi/installing-azure-customizations.html#installation-azure-config-yaml_installing-azure-customizations[Sample customized install-config.yaml file for Azure] for a sample configuration file, or set a non-null parameter as in the following example:
```
Comment (tmalove) [inline release_notes/ocp-4-17-release-notes.adoc:1886]: The link should be an 'href' for an HTML file or a .adoc file for and 'xref', yes? It did resolve though.

Ascii writer's guide: https://asciidoctor.org/docs/asciidoc-writers-guide/#cross-references

**Review by tmalove — APPROVED** [review]

**Review by tmalove — APPROVED** [review]

File: release_notes/ocp-4-17-release-notes.adoc (hunk 2 above)
Comment (bscott-rh) [reply release_notes/ocp-4-17-release-notes.adoc]: Good catch, I switched it to .adoc

File: release_notes/ocp-4-17-release-notes.adoc (hunk 1 above)
Comment (gpei) [reply release_notes/ocp-4-17-release-notes.adoc]: @bscott-rh seems it's not well updated, would you mind taking another look?

File: release_notes/ocp-4-17-release-notes.adoc [hunk 3]
```diff
@@ -1890,6 +1890,17 @@ In the following tables, features are marked with the following statuses:
 
 * The DNF package manager included in {op-system-first} images cannot be used at runtime, because DNF relies on additional packages to access entitled nodes in a cluster that are under a Red Hat subscription. As a workaround, use the `rpm-ostree` command instead. (link:https://issues.redhat.com/browse/OCPBUGS-35247[*OCPBUGS-35247*])
 
+* When installing a cluster on {azure-full}, the installation will fail if no `install-config.yaml` file is provided or if the `controlPlane.platform.azure` stanza is null in the provided `install-config.yaml` file. (link:https://issues.redhat.com/browse/OCPBUGS-42296[*OCPBUGS-42296*])
```
Comment (gpei) [inline release_notes/ocp-4-17-release-notes.adoc:1893]:
```suggestion
* When installing a cluster on {azure-full}, the installation will fail if no `install-config.yaml` file is provided or if the `controlPlane.platform` stanza is null in the provided `install-config.yaml` file. (link:https://issues.redhat.com/browse/OCPBUGS-42296[*OCPBUGS-42296*])
```

Comment (sadasu) [reply release_notes/ocp-4-17-release-notes.adoc:1893]: In the code, we are treating not setting `controlPlane.platform.azure` stanza as null.

How about :
 `When installing a cluster on {azure-full}, the installation will fail if no `install-config.yaml` file is provided or if the **`controlPlane.platform.azure` stanza is not set** in the provided `install-config.yaml` file. (link:https://issues.redhat.com/browse/OCPBUGS-42296[*OCPBUGS-42296*])
`

**Pull request merged**